lib/message_verifier.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class MessageVerifier
+  class << self
+    def encode(payload, exp = 5.minutes.from_now)
+      payload[:exp] ||= exp
+      verifier.generate payload
+    end
+
+    def decode(token)
+      body = verifier.verify(token)
+
+      raise ActiveSupport::MessageVerifier::InvalidSignature if Time.current > body[:exp]
```

Comment:
Whats the point of raising here, then rescuing and throwing away the error?

Reply:
`body = verifier.verify(token)` returns the same error if the signature is invalid, and that line is to make sure that the signing has not expired, so that we can cleanly return nil if the token is either tampered with or expired. 

Although, it might be better practice to put everything into a `begin... rescue` block.

Reply:
Yea was raising the same error instead of adding a separate ‘return nil’. Is raising an error and then catching it bad practice?

Reply:
Ah I see. An expired token doesn't seem like an invalid signature error, could we make a new error class `TokenExpired` or something and then raise that, then in the rescue, catch both? We can just return nil in the rescue, but this lets us do something different depending on how we failed later on.